messaging/src/main/java/org/axonframework/commandhandling/SimpleCommandHandlingComponent.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2010-2025. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.commandhandling;
+
+import jakarta.annotation.Nonnull;
+import org.axonframework.messaging.MessageStream;
+import org.axonframework.messaging.QualifiedName;
+import org.axonframework.messaging.unitofwork.ProcessingContext;
+
+import java.util.Objects;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * TODO This should be regarded as a playground object to verify the API. Feel free to remove, adjust, or replicate this class to your needs.
+ *
+ * @author Steven van Beelen
+ * @since 5.0.0
+ */
+public class SimpleCommandHandlingComponent implements CommandHandlingComponent {
+
+    private final ConcurrentHashMap<QualifiedName, CommandHandler> commandHandlers;
+
+    public SimpleCommandHandlingComponent() {
+        this.commandHandlers = new ConcurrentHashMap<>();
+    }
+
+    @Nonnull
+    @Override
+    public MessageStream<? extends CommandResultMessage<?>> handle(@Nonnull CommandMessage<?> command,
+                                                                   @Nonnull ProcessingContext context) {
+        QualifiedName name = command.type().qualifiedName();
+        // TODO add interceptor knowledge
+        CommandHandler handler = commandHandlers.get(name);
+        if (handler == null) {
+            // TODO this would benefit from a dedicate exception
+            return MessageStream.failed(new IllegalArgumentException(
```

Comment:
How do you make the decisions which one to use:
`MessageStream.failed` or message stream with `GenericCommandResultMessage` (Exception inside).

Reply:
The intent is to move to the failed message stream instead of failed messages, @MateuszNaKodach!
So, all the exception stuff on result message should, eventually, be removed for AF5

Reply:
Thanks! :) Ah yes, we were talkin about that while thinking how to handle MessageType resolver for Exception. I forgot! :)